src/commands/yubihsm/setup.rs

```diff
@@ -41,14 +49,45 @@ pub struct SetupCommand {
     /// Print debugging information
     #[options(short = "v", long = "verbose")]
     pub verbose: bool,
+
+    /// Only display derived keys - do not reinitialize HSM
+    #[options(short = "d", long = "derive-only")]
+    pub derive_only: bool,
+
+    /// Restore an HSM from an existing 24-word remonic
+    #[options(short = "r", long = "restore")]
+    pub restore: bool,
+
+    /// Write a provisioning report as JSON to the given filename
+    #[options(short = "w", long = "write-report")]
+    pub write_report: Option<PathBuf>,
 }
 
 impl Callable for SetupCommand {
     /// Perform initial YubiHSM dervice provisioning
     fn call(&self) {
         let hsm_connector = crate::yubihsm::connector();
+        let hsm_serial_number = get_hsm_client(&hsm_connector)
+            .device_info()
+            .expect("error getting device info")
+            .serial_number;
+
+        let mnemonic = if self.restore {
+            println!("Restoring and reprovisioning YubiHSM from existing 24-word mnemonic phrase.");
+            println!();
+
+            let mnemonic_phrase = rustyline::Editor::<()>::new()
```

Comment:
Took a look at a few different options for this...

There's [rpassword](https://crates.io/crates/rpassword) for reading passwords without displaying them, however for entering a 24-word phrase I think it's pretty nice if you can see what you're typing.

The KISS solution is [io::stdin().read_line()](https://doc.rust-lang.org/std/io/struct.Stdin.html#method.read_line) but that isn't "correctable" in the event something is mistyped. It may not be too difficult to parse the string and look for corrections, however.

`rustyline` has nice UX but seems like complete overkill.